roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,36 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set gateway address for multi subnet 
+  set_fact: dfl_nw_gw="{{ item.gateway }}"
+  when: (update_routes|default('false')|bool) and ((ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') ))) 
+  with_items: "{{ undercloud_cidr }}"
```

Comment:
@j2sol Don't we want to throw an error if gateway is not set and update_routes is set to yes?
Right now, the code will fail if the user tries to update the routes without setting gateway in all.yml.